hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/SelectProcessorSupplier.java

```diff
@@ -0,0 +1,195 @@
+                                   List<Expression<?>> projection) {
+        this.jdbcUrl = requireNonNull(jdbcUrl, "jdbcUrl must not be null");
+        this.tableName = requireNonNull(tableName, "tableName must not be null");
+        this.fields = requireNonNull(fields, "fields must not be null");
+        this.predicate = predicate;
+        this.projection = projection;
+    }
+
+    @Override
+    public void init(@Nonnull Context context) throws Exception {
+        evalContext = ExpressionEvalContext.from(context);
+        query = buildQuery();
+    }
+
+    private String buildQuery() {
+        ExpressionTranslator translator = new ExpressionTranslator(evalContext, fields);
+        String select;
+        if (projection != null) {
+            StringBuilder sb = new StringBuilder();
+            for (int i = 0; i < projection.size(); i++) {
+                Expression<?> projectionExpression = projection.get(i);
+                String translated = translator.translate(projectionExpression);
+                sb.append(translated);
+                if (i < projection.size() - 1) {
+                    sb.append(",");
+                }
+            }
+            select = sb.toString();
+        } else {
+            select = "*";
+        }
+        if (predicate != null) {
+            String wherePredicate = translator.translate(predicate);
+            return "SELECT " + select + " FROM " + tableName + " WHERE " + wherePredicate;
+        } else {
+            return "SELECT " + select + " FROM " + tableName;
+        }
+    }
+
+    @Nonnull
+    @Override
+    public Collection<? extends Processor> get(int count) {
+        List<Processor> processors = new ArrayList<>(count);
+        for (int i = 0; i < count; i++) {
+            Processor processor = new ReadJdbcP<>(
+                    () -> DriverManager.getConnection(jdbcUrl),
+                    (connection, parallelism, index) -> {
+                        PreparedStatement statement = connection.prepareStatement(query);
+                        try {
+                            return statement.executeQuery();
+                        } catch (SQLException e) {
+                            statement.close();
+                            throw e;
+                        }
+                    },
+                    rs -> {
+                        int columnCount = rs.getMetaData().getColumnCount();
+                        Object[] row = new Object[columnCount];
+                        for (int j = 0; j < columnCount; j++) {
+                            Object value = rs.getObject(j + 1);
+                            row[j] = convertValue(value);
+                        }
+
+                        return new JetSqlRow(evalContext.getSerializationService(), row);
+                    }
+            );
+            processors.add(processor);
+        }
+        return processors;
+    }
+
+    private Object convertValue(Object value) {
```

Comment:
What about enums, do we support any automatic conversion?

Reply:
This conversion is from objects returned by the JDBC driver to what we support in our SQL. I don't think we need any special handling for enums here.